src/main/payment/main/G1HRPS/Payment.java

```diff
@@ -1,164 +1,277 @@
 	 * 
 	 * @param guest_id
 	 */
-	public void SetGuestId(int guest_id) {
-		// TODO - implement Payment.SetGuestId
 
-		this.guest_id_ = guest_id;
+	public void SetGuestID(String guest_id) {
+
+		guest_id_ = guest_id;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetDiscount(float discount) {
-		// TODO - implement Payment.SetDiscount
+	/**
+	 * 
+	 * @return	Guest ID of guest that made payment
+	 */
+
+	public String GetGuestID() {
 
-		this.discounts_ = discount;
+		return guest_id_;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetRoomCharges() {
-		// TODO - implement Payment.SetRoomCharges
+	/**
+	 * Assigns new room_num to this payment
+	 * 
+	 * @param room_num
+	 */
+
+	public void SetRoomNum(int room_num) {
+
+		room_num_ = room_num;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetRoomServices() {
-		// TODO - implement Payment.SetRoomServices
+	/**
+	 * 
+	 * @return	room number of this payment
+	 */
+
+	public int GetRoomNum() {
 
-		throw new UnsupportedOperationException();
+		return room_num_;
+		
 	}
 
 	/**
+	 * Assigns new discount rate to this payment
 	 * 
-	 * @param Tax
+	 * @param discount
 	 */
-	public void SetTax(float Tax) {
-		// TODO - implement Payment.SetTax
 
-		this.tax_ = Tax;
+	public void SetDiscount(float discount) {
+
+		discounts_ = discount;
 
-		throw new UnsupportedOperationException();
 	}
 
 	/**
 	 * 
-	 * @param bill_total
+	 * @return	discount rate of this payment
 	 */
-	public void SetBillTotal(float bill_total) {
-		// TODO - implement Payment.SetBillTotal
 
-		this.bill_total_ = bill_total;
+	public float GetDiscount() {
+
+		return discounts_;
 
-		throw new UnsupportedOperationException();
 	}
 
 	/**
+	 * Assigns room charge of this payment
 	 * 
-	 * @param status
+	 * @param room_charges
 	 */
-	public void SetStatus(PaymentStatus status) {
-		// TODO - implement Payment.SetStatus
 
-		this.status_ = status;
+	public void SetRoomCharges(float room_charges) {
+
+		room_charges_ = room_charges;
+
+	}
+
+	/**
+	 * 
+	 * @return	room charge of this payment
+	 */
+
+	public float GetRoomCharges() {
+		
+		return room_charges_;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public int GetGuestId() {
-		// TODO - implement Payment.GetGuestId
+	/**
+	 * Assign room service charge of this payment 
+	 * 
+	 * @param rs_charges The new room service charge to be assigned
+	 */
 
-		return guest_id_;
+	public void SetRsCharges(float rs_charges) {
+
+		rs_charges_ = rs_charges;
 
-		// throw new UnsupportedOperationException();
 	}
 
-	public float GetDiscount() {
-		// TODO - implement Payment.GetDiscount
+	/**
+	 * Calculates room service charge of a room
+	 * from each of the room services in the list
+	 * with a matching room number
+	 * 
+	 * @param room_service_order_list
+	 */
 
-		return this.discounts_;
+	public void SetRsCharges(List<RoomServiceOrder> room_service_order_list) {
```

Comment:
I don't think this is very efficient in the sense that if there are a lot of room services from many different guests and rooms, each time you want to calculate this, you have to go through the whole list. The provided list should already be filtered to relate to only this guest, looking at our miro board, we search for a guest to check out, then print bill info for said guest.

This can be just a float value. RoomServiceManager should have a getlist function that provides a list ofRoom Service Orders by guest.